lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -187,7 +192,12 @@ export class PublicClientApplication extends ClientApplication implements IPubli
             this.logger.verbose("acquireTokenSilent - attempting to acquire token from web flow");
             const silentCacheClient = this.createSilentCacheClient(request.correlationId);
             const silentRequest = await silentCacheClient.initializeSilentRequest(request, account);
-            result = silentCacheClient.acquireToken(silentRequest).catch(async () => {
+            result = silentCacheClient.acquireToken(silentRequest).catch(async (error) => {
+                if (request.silentTokenRetrievalStrategy === SilentTokenRetrievalStrategy.CacheOnly) {
+                    this.logger.error("SilentTokenRetrievalStrategy is set to CacheOnly, not attempting to renew token silently.");
+                    throw error;
```

Comment:
Do we want to throw the raw cache exception or map this to something more friendly?  I'm assuming right now if we throw a cache error today we'd catch that (it would never get returned) and if the subsequent network request failed with a network or server error we'd throw that back.

Reply:
I think we would either get [this error](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/eb612862f60dd6b9d28e6bec8e08989890dbf441/lib/msal-common/src/error/ClientAuthError.ts#L161-L164) (the error message for which should get updated to reflect this new scenario @Robbie-Microsoft), or an interaction_required/login_required error from the server, which I think is fine.